### PR TITLE
fix(create-waku): do not download from main

### PR DESCRIPTION
**Pull request by dai-shi**
As I said in #581, downloading from `main` will break.

- [x] show help
- [x] download from the released branch
- [ ] refactor

### CODE REVIEW

File: packages/create-waku/src/index.ts [hunk 1]
```diff
@@ -185,13 +199,13 @@ async function init() {
     process.exit(1);
   }
   let repoInfo: RepoInfo | undefined;
-  const example = getExample(tokens);
 
-  if (example) {
+  // FIXME this is pretty hard to follow. Separate functions?
```
Comment (dai-shi) [inline packages/create-waku/src/index.ts:203]: @ojj1123 I found the code pretty hard to read. There is a big room for improvement. I will factor out the functionality into the separate file, and you can consider some refactors there. Please work on refactors and adding tests before adding new capabilities.

File: packages/create-waku/src/helpers/example-option.ts [hunk 2]
```diff
@@ -116,15 +118,130 @@ export async function downloadAndExtractRepo(
...
     await downloadTarStream(
-      'https://codeload.github.com/dai-shi/waku/tar.gz/main',
+      `https://codeload.github.com/dai-shi/waku/tar.gz/${ref}`,
     ),
     tar.x({
       cwd: root,
       strip: 2 + name.split('/').length,
-      filter: (p) => p.includes(`waku-main/examples/${name}/`),
+      filter: (p) =>
+        p.replace(/waku-[^/]+\//, '').includes(`examples/${name}/`),
```
Comment (dai-shi) [inline packages/create-waku/src/helpers/example-option.ts:134]: This seems too hacky. Any idea to improve?

File: packages/create-waku/src/helpers/example-option.ts [hunk 3]
```diff
@@ -116,15 +118,130 @@ export async function downloadAndExtractRepo(
...
+    } else {
+      console.log(
+        `Downloading files for example ${cyan(example)}. This might take a moment.`,
+      );
+      console.log();
+      await downloadAndExtractExample(root, example, defaultRef);
+    }
+  } catch (err) {
+    // download error
+    throw new Error(isErrorLike(err) ? err.message : err + '');
```
Comment (dai-shi) [inline packages/create-waku/src/helpers/example-option.ts:241]: It doesn't seem doing anything good. If we want friendly error message we shouldn't throw anyway.

**Comment by dai-shi** [comment]
> I will factor out the functionality into the separate file

I think I did more than that.

**Review by ojj1123 — COMMENTED** [review]
thanks to you, the `index.ts` is simple and getting the high readability.
I will write test code first as you said

File: packages/create-waku/src/helpers/example-option.ts (hunk 3 above)
Comment (ojj1123) [reply packages/create-waku/src/helpers/example-option.ts]: nice your catch! I agree with you :)

File: packages/create-waku/src/index.ts [hunk 4]
```diff
@@ -1,145 +1,57 @@
...
-  existsInRepo,
-  getRepoInfo,
-  hasRepo,
-  type RepoInfo,
-} from './helpers/example-option';
+  parseExampleOption,
+  downloadAndExtract,
+} from './helpers/example-option.js';
+
+const DEFAULT_REF = 'v0.20.0-alpha.2';
```
[the remark this answers is not in the document]
Comment (dai-shi) [reply packages/create-waku/src/index.ts:16]: yes, manually.

File: packages/create-waku/src/index.ts [hunk 5]
```diff
@@ -1,145 +1,57 @@
...
     example: {
       type: 'string',
     },
+    help: {
+      type: 'boolean',
+      short: 'h',
+    },
   },
-  tokens: true,
 });
```
[the remark this answers is not in the document]
Comment (dai-shi) [reply packages/create-waku/src/index.ts:30]: Let's leave it. It's the same as what we do in `waku` too.

File: packages/create-waku/src/helpers/example-option.ts [hunk 6]
```diff
@@ -116,15 +118,113 @@ export async function downloadAndExtractRepo(
...
+  repoInfo?: RepoInfo;
+};
+
+export async function parseExampleOption(
+  example: string | undefined,
+  defaultRef: string,
+): Promise<ParsedExampleOption | undefined> {
+  if (!example) {
+    return undefined;
+  }
```
[the remark this answers is not in the document]
Comment (dai-shi) [reply packages/create-waku/src/helpers/example-option.ts:163]: yeah, but let's leave it as type says `undefined` instead of `void`. or, we can return null.

File: packages/create-waku/src/helpers/example-option.ts (hunk 2 above)
Comment (dai-shi) [reply packages/create-waku/src/helpers/example-option.ts:134]: it has both replace and includes. `name` can contain `waku-foo`.

File: packages/create-waku/src/helpers/install-template.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  packageName: string;
+}) {
+  const pkg = {
+    name: packageName,
+    version: '0.0.0',
+  };
+
+  const templateRoot = path.join(
+    fileURLToPath(import.meta.url),
+    '../../template',
```
Comment (dai-shi) [inline packages/create-waku/src/helpers/install-template.ts:21]: It's very confusing that this isn't `../../../template`.
